### PR TITLE
bump(k8s) 1.21.0

### PR DESCRIPTION
**Pull request by gabemontero**
based on announcement from Maciej in http://mailman-int.corp.redhat.com/archives/aos-devel/2021-April/msg00130.html

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,9 @@
 module github.com/openshift/cluster-samples-operator
 
-go 1.13
+go 1.16
 
 require (
-	github.com/NYTimes/gziphandler v1.1.1 // indirect
```
Comment (gabemontero) [inline go.mod:6]: resulted from go mod tidy

File: go.mod [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
 	github.com/emicklei/go-restful v2.11.1+incompatible // indirect
-	github.com/go-logr/logr v0.3.0 // indirect
```
Comment (gabemontero) [inline go.mod:8]: resulted from go mod tidy

File: go.mod [hunk 3]
```diff
@@ -17,11 +15,11 @@ require (
 	github.com/prometheus/client_golang v1.7.1
 	github.com/prometheus/client_model v0.2.0
 	github.com/prometheus/common v0.10.0
-	github.com/sirupsen/logrus v1.6.0
+	github.com/sirupsen/logrus v1.7.0
```
Comment (gabemontero) [inline go.mod:18]: resulted from go mod tidy

**Comment by gabemontero** [comment]
unrelated `: [sig-instrumentation][Late] Alerts shouldn't report any alerts in firing or pending state apart from Watchdog and AlertmanagerReceiversNotConfigured and have no gaps in Watchdog firing [Suite:openshift/conformance/parallel]` on image eco

/test e2e-aws-image-ecosystem

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**